src/hooks/api/useSplits.ts

```diff
@@ -184,34 +197,125 @@ export function useAccountsTotal(
  */
 export function useAccountsMonthlyTotal(
   interval?: Interval,
-): UseQueryResult<AccountsMonthlyTotals> {
+): UseQueryResult<AccountsTotals[]> {
   interval = interval || Interval.fromDateTimes(
     DateTime.now().minus({ month: 6 }).startOf('month'),
     DateTime.now(),
   );
   const { data: accounts, dataUpdatedAt: accountsUpdatedAt } = useAccounts();
   const { data: prices, dataUpdatedAt: pricesUpdatedAt } = usePrices({});
 
+  const aggregate = React.useCallback(
+    ((data: AccountsTotals[]) => {
+      const dates = (interval as Interval).splitBy({ month: 1 }).map(d => (d.start as DateTime).endOf('month'));
+      dates[dates.length - 1] = (interval as Interval).end as DateTime;
+      return data.map((d, i) => aggregateChildrenTotals(
+        'type_root',
+        accounts as Account[],
+        prices as PriceDBMap,
+        dates[i],
+        d,
+      ));
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [accountsUpdatedAt, pricesUpdatedAt, interval.toISODate()],
+  );
+
   const queryKey = [
     ...Split.CACHE_KEY,
     {
       aggregation: 'monthly-total',
       dates: interval.toISODate(),
       accountsUpdatedAt,
-      pricesUpdatedAt,
     },
   ];
   const result = useQuery({
     queryKey,
     queryFn: fetcher(
       () => getMonthlyTotals(
         accounts as Account[],
-        prices as PriceDBMap,
         interval,
       ),
       queryKey,
     ),
-    enabled: !!accounts && !!prices,
+    enabled: !!accounts,
+    select: aggregate,
+  });
+
+  return result;
+}
+
+/**
+ * Aggregates monthly splits for each account accumulating the value for each month.
+ * This is useful to produce monthly worth histograms.
+ *
+ * For accounts where their children
+ * have different commodity, the monthly aggregation is converted using an
+ * exchange rate for each respective month
+ */
+export function useAccountsMonthlyWorth(
+  interval?: Interval,
+): UseQueryResult<AccountsTotals[]> {
+  interval = interval
+    || Interval.fromDateTimes(
+      DateTime.now().minus({ month: 6 }).startOf('month'),
+      DateTime.now(),
+    );
+  const { data: accounts, dataUpdatedAt: accountsUpdatedAt } = useAccounts();
+  const { data: prices, dataUpdatedAt: pricesUpdatedAt } = usePrices({});
+  const { data: totals, dataUpdatedAt: totalsUpdatedAt } = useAccountsTotals(
+    (interval.start as DateTime).endOf('month') as DateTime,
+    (data) => data,
+  );
+
+  const aggregate = React.useCallback(
```

Comment:
This one is similar to the monthlyTotal above EXCEPT that for the first month, we retrieve the total for the accounts using `useAccountsTotals` and then we set it as the first month data. We get monthly totals then only for interval.start + 1 to interval.end